### ISSUE TITLE
No celestial bodies after a while, no game over either

### ISSUE THREAD

**Issue by Oyzuu** [kirub/gmtk-19#3]

**Comment by MaxiPoki** [comment]
Supernova should be quicker to catch up with you also... To be able to game over quicker...
Maybe check if we out of bounds from the map, force the supernova size to near the player and let it grow to "force" the game over.

**Comment by kirub** [comment]
Actually several solutions there (now that we have more time). 
Max suggestion is one of them, but we could also generating the map on the fly.
The way I created the "high density map" was placing some planets in a limited squared area then duplicating them.
In a similar way we could then making this squared area a prefab, then instantiating it in code (with maybe some changes in orientation ).
We could even stream having only planets around the player and stream them according to its direction.

**Comment by Oyzuu** [comment]
Could randomize an important number of planet positions as (x, y) tuples and only instantiating them near player, drawing from an item pool to prevent loading too much in RAM. (10 planets max maybe ?)

**Comment by kirub** [comment]
Yep, there's 3 problems with that is making sure they don't overlap but that's pretty easy to do and the helpers saying where the planets are: we need to pre-stream enough planets. The last issue would be going back but it's manageable as well (just need to keep tracks of planet generation events and generating a planet should be deterministic so we could "rewind" the streaming).

**Comment by Oyzuu** [comment]
Planet helpers could point to an already existing Vector, even if the planet is not instantiated already. Radial generation could help to reproduce the feeling that we're fleeing out of from the center of the galaxy, meeting less and less bodies as we fly away.
